src/web/releases.rs

```diff
@@ -273,75 +273,61 @@ fn get_search_results(
-            -- crates that don't match the query closely enough
-            WHERE
-                latest_release.rustdoc_status
-                AND NOT latest_release.yanked
-                AND (
-                    -- Crates names that match the query sandwiched between wildcards will pass
-                    crates.name ILIKE CONCAT('%', $1, '%')
-                    -- Crate names with which the levenshtein distance is closer or equal to 3 will pass
-                    OR levenshtein_less_equal($1, crates.name, 3) <= 3
-                    -- Crates where their content matches the query will pass
-                    OR plainto_tsquery($1) @@ crates.content
-                )
-            
-            GROUP BY crates.id, releases.id
-            
-            -- Ordering is prioritized by how closely the query matches the name, how closely the
-            -- query matches the description, and finally how many downloads the crate has
-            -- NOTE: this means that exact matches will be shown first
-            ORDER BY distance DESC,
-                content_rank DESC,
-                downloads_total DESC
-            
-            -- Allows pagination
-            LIMIT $2 OFFSET $3";
+    let statement = "
+        SELECT
+            crates.name,
+            releases.version,
+            releases.description,
+            releases.target_name,
+            releases.release_time,
+            releases.rustdoc_status,
+            SUM(crates.github_stars),
+            COUNT(*) OVER() as total
+        FROM releases
+        INNER JOIN crates on releases.crate_id = crates.id
+        WHERE
+            -- Only select the newest release by release time
+            releases.id = (
+                SELECT releases.id
+                FROM releases
+                -- Filter unbuilt/failing builds and yanked releases
+                WHERE
+                    releases.crate_id = crates.id
+                    AND releases.rustdoc_status
+                    AND NOT releases.yanked
+                ORDER BY releases.release_time DESC
+                LIMIT 1
+            )
+            -- Only select releases/crates that pass our criteria:
+            --      - Levenshtein distance if the name and query is greater than three
+            --      - The query sandwiched between wildcards matches the crate's name
+            --      - The query matches the release's description
+            AND (
+                levenshtein_less_equal($1, crates.name, 3) <= 3
+                OR crates.name ILIKE CONCAT('%', $1, '%')
+                OR plainto_tsquery($1) @@ to_tsvector(releases.description)
+            )
+        GROUP BY crates.id, releases.id
+        -- Order by the levenshtein distance of the name, the text search ranking of the description
+        -- and finally the number of downloads
+        ORDER BY
+            levenshtein_less_equal($1, crates.name, 3) ASC,
```

Comment:
I think this should also take into account the `ILIKE` so that e.g. `regex-syntax` will come before a crate with an unrelated name but related description.

Reply:
Do you know of a way to do that?

Reply:
Same thing you're selecting by:
```suggestion
            levenshtein_less_equal($1, crates.name, 3) ASC,
            crates.name ILIKE CONCAT('%', $1, '%'),
```